### PR TITLE
Add template for data log

### PR DESCRIPTION
**Pull request by serenayozalp**

### CODE REVIEW

**Comment by phwrwalms** [comment]
Thanks @serenayozalp! 

How do you anticipate this being used in the workflow? Would it be filled out once a week, say? 

I really like the idea of having structured documentation of what we find on projects, and it makes sense what's included here. I'm less sure about whether it should be incorporated into phwcookiecutter or live elsewhere. Reason being currently phwcookiecutter is very generic, so suitable for both package-like projects (e.g. phwplot) and analysis projects (e.g. betsidiabetes). It feels like this would make it more specialised to the latter? Which is fine, if that's the direction we want to take, but keen to be intentional about that change. Also, currently phwcookiecutter is suitable for both our projects and other people's (within PHW) - the reference to the shared drive seems to make it just for us?

**Comment by serenayozalp** [comment]
> Thanks @serenayozalp!
> 
> How do you anticipate this being used in the workflow? Would it be filled out once a week, say?
> 
> I really like the idea of having structured documentation of what we find on projects, and it makes sense what's included here. I'm less sure about whether it should be incorporated into phwcookiecutter or live elsewhere. Reason being currently phwcookiecutter is very generic, so suitable for both package-like projects (e.g. phwplot) and analysis projects (e.g. betsidiabetes). It feels like this would make it more specialised to the latter? Which is fine, if that's the direction we want to take, but keen to be intentional about that change. Also, currently phwcookiecutter is suitable for both our projects and other people's (within PHW) - the reference to the shared drive seems to make it just for us?

Thanks @phwrwalms. I've been thinking about creating a collection of templates in the repository, including issue templates, PR templates, and others. Based on the type of project generated by the cookiecutter, users can decide whether or not they want to use them. I envision offering a range of templates, so people can choose what best suits their needs. If these templates are readily available, it will likely encourage better documentation practices, as many projects currently lack structured, consistent documentation. If this sounds good to you, I'll gather all the templates into a "templates" folder before merging.

**Comment by phwrwalms** [comment]
I like this idea. 

But maybe use if for other things but don't move the issue or the PR templates just yet as GitHub uses where they sit to find them and offer them up when you do PRs/ issues :) 

And for these other things, I think there is a way to get the cookiecutter code to give you some options in running. So it could ask you to "do you want to include x, y, z" which would be nice (possibly related to your issue #7). If that sounds good I can add to the backlog.

**Comment by phwrwalms** [comment]
Or, another option is, we have 2 separate things: 
- this, which is just a cookiecutter for a repo with 1 opinionated setup 
- then a separate repo with loads of different templates 

I don't know which I prefer ❓

**Comment by serenayozalp** [comment]
> I like this idea.
> 
> But maybe use if for other things but don't move the issue or the PR templates just yet as GitHub uses where they sit to find them and offer them up when you do PRs/ issues :)
> 
> And for these other things, I think there is a way to get the cookiecutter code to give you some options in running. So it could ask you to "do you want to include x, y, z" which would be nice (possibly related to your issue #7). If that sounds good I can add to the backlog.

That's great idea. Let me have a look _giving option_ functionality before merge then :)

**Review by phwjrees — APPROVED** [review]
Have tested this and it all works as intended. Approved :)

File: {{cookiecutter.project_repo_name}}/.github/weekly_data_exp_log_template.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+YYYY-MM-DD_weekly_log_[team_member_name].md
+
+
+### **Saving Instructions:**
+Save the Markdown file in the shared drive at the following location:
+
+/data/docs/ in shared drive
+
+
```
Comment (phwrwalms) [inline {{cookiecutter.project_repo_name}}/.github/weekly_data_exp_log_template.md]: Should we now get this translated?

File: {{cookiecutter.project_repo_name}}/.github/templates/{% if cookiecutter.include_user_story_templates == 'yes' %}mater-stori-defnyddiwr-safonol.md{% endif %}
Comment (phwrwalms) [inline {{cookiecutter.project_repo_name}}/.github/templates/{% if cookiecutter.include_user_story_templates == 'yes' %}mater-stori-defnyddiwr-safonol.md{% endif %}:1]: @serenayozalp I'm not sure these are ending up in the right place if you respond "yes". 

- I think you end up with 2 copies of the template, one in its original place and one in this templates folder. So I think you can delete the original (or this one) in the file structure. Also, the original one will always appear, even if you select "no"
- For the issue templates, if you want them to appear on GitHub, I think they need to sit in their original location of ISSUE_TEMPLATE. So could you move this file (with the conditional filename) to .github/ISSUE_TEMPLATE?

File: cookiecutter.json [hunk 2]
```diff
@@ -1,7 +1,9 @@
...
+    "project_name": "Your new project name - Enw eich prosiect newydd",
     "project_repo_name": "{{ cookiecutter.project_name.lower().replace(' ', '_') }}",
-    "project_description": "Brief project description/Disgrifiad byr o'r prosiect",
-    "project_team": "Project team name e.g. DKR Data Science/Enw tîm y prosiect e.e. Gwyddor Data",
-    "contact_email": "Project contact email address/Cyfeiriad e-bost cyswllt y prosiect",
+    "project_description": "Brief project description - Disgrifiad byr o'r prosiect",
+    "project_team": "Project team name e.g. DKR Data Science - Enw tîm y prosiect e.e. Gwyddor Data",
+    "contact_email": "Project contact email address - Cyfeiriad e-bost cyswllt y prosiect",
+    "include_user_story_templates": "yes/no",
+    "include_weekly_data_exp_log_template": "yes/no"
```
Comment (phwrwalms) [inline cookiecutter.json]: really the question + choice offered here should be bilingual. Maybe for the choice an easy way round this is to make it 1/0 instead?

File: {{cookiecutter.project_repo_name}}/.github/templates/{% if cookiecutter.include_weekly_data_exp_log_template == 'yes' %}weekly-data-exp-log-template.md{% endif %} [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+### **File Naming Convention:**
+Save this file using the following naming convention:
+
+YYYY-MM-DD_weekly_log_[team_member_name].md
+
+
+### **Saving Instructions:**
+Save the Markdown file in the shared drive at the following location:
+
+/data/docs/ in shared drive
```
Comment (phwrwalms) [inline {{cookiecutter.project_repo_name}}/.github/templates/{% if cookiecutter.include_weekly_data_exp_log_template == 'yes' %}weekly-data-exp-log-template.md{% endif %}]: This is interesting - the idea is the md wouldn't be committed to the repo? Wouldn't you want to keep this info in the repo? Or is it in case of sensitive data?

**Review by phwrwalms — COMMENTED** [review]
This is really nice additional functionality. It basically works for me, but I've suggested some changes to where files sit, and I think we should now also be able to delete files that are obsolete. 

We also need to think about translation for new content - I'm not quite sure at which point this should happen, and it's not so important while it's not public. In this case, I think we should merge and then get this extra bit translated, but we should add an issue at least.

File: {{cookiecutter.project_repo_name}}/.github/weekly_data_exp_log_template.md (hunk 1 above)
Comment (serenayozalp) [reply {{cookiecutter.project_repo_name}}/.github/weekly_data_exp_log_template.md]: > Should we now get this translated?

Yes, we will do that. I added in our translation list

File: {{cookiecutter.project_repo_name}}/.github/templates/{% if cookiecutter.include_weekly_data_exp_log_template == 'yes' %}weekly-data-exp-log-template.md{% endif %} (hunk 3 above)
Comment (serenayozalp) [reply {{cookiecutter.project_repo_name}}/.github/templates/{% if cookiecutter.include_weekly_data_exp_log_template == 'yes' %}weekly-data-exp-log-template.md{% endif %}]: > This is interesting - the idea is the md wouldn't be committed to the repo? Wouldn't you want to keep this info in the repo? Or is it in case of sensitive data?

I believe there is a potential risk of including business-sensitive data when logging our experiments in the current template. To address this, my initial idea was to store these logs where we store our datasets. Based on this, I added a "Saving instructions" section to the template, which specifies the directory (e.g., /data/docs/) where these logs are stored for each project. This will give us a clear reference point on where to find the relevant logs.

File: cookiecutter.json (hunk 2 above)
Comment (serenayozalp) [reply cookiecutter.json]: > really the question + choice offered here should be bilingual. Maybe for the choice an easy way round this is to make it 1/0 instead?

Even if we choose to go with the numbered version, we should still include comments or explanations indicating that 1 means "yes" and 0 means "no," to provide clarity. I will edit this adding translation in place

File: {{cookiecutter.project_repo_name}}/.github/templates/{% if cookiecutter.include_user_story_templates == 'yes' %}mater-stori-defnyddiwr-safonol.md{% endif %}
Comment (serenayozalp) [reply {{cookiecutter.project_repo_name}}/.github/templates/{% if cookiecutter.include_user_story_templates == 'yes' %}mater-stori-defnyddiwr-safonol.md{% endif %}:1]: I moved the issue template where it was. But I dont understand which file we ended up having 2 copies. Please test it again after my last commit and let me know what I am missing :)

File: {{cookiecutter.project_repo_name}}/.github/templates/{% if cookiecutter.include_weekly_data_exp_log_template == 'yes' %}weekly-data-exp-log-template.md{% endif %} (hunk 3 above)
Comment (phwrwalms) [reply {{cookiecutter.project_repo_name}}/.github/templates/{% if cookiecutter.include_weekly_data_exp_log_template == 'yes' %}weekly-data-exp-log-template.md{% endif %}]: Makes sense :)

File: cookiecutter.json (hunk 2 above)
Comment (phwrwalms) [reply cookiecutter.json]: Thanks 👍

File: {{cookiecutter.project_repo_name}}/.github/templates/{% if cookiecutter.include_user_story_templates == 'yes' %}mater-stori-defnyddiwr-safonol.md{% endif %}
Comment (phwrwalms) [reply {{cookiecutter.project_repo_name}}/.github/templates/{% if cookiecutter.include_user_story_templates == 'yes' %}mater-stori-defnyddiwr-safonol.md{% endif %}:1]: So for example, the file standard-user-story-issue.md now appears in three places in the folder structure. I think that means 2 or 3 (depending if the answer 0 or 1 is given to the prompt) copies of it will appear in the corresponding place in the folder you make using the cookiecutter. These locations are: 
image

[You can confirm this by trying to apply the cookiecutter, as per the instructions but modified to use the relevant branch: it will create a repo with this file in 3 locations.]

I think instead if you move the files around as follows it should work: 
image

Then also: 
-  the same thing for the Welsh language user story
- the data log template can stay where it is BUT delete the second copy of it (the one without the conditional name) 

I hope this makes sense, but if it doesn't let's talk about it verbally as it's a really quick change but hard to put into words :)

Comment (serenayozalp) [reply {{cookiecutter.project_repo_name}}/.github/templates/{% if cookiecutter.include_user_story_templates == 'yes' %}mater-stori-defnyddiwr-safonol.md{% endif %}:1]: @phwrwalms, interesting! I just run the code and it is not happening to me. Lets check it on Monday together

File: cookiecutter.json [hunk 4]
```diff
@@ -1,7 +1,8 @@
...
-    "project_name": "Your new project name/ Enw eich prosiect newydd",
+    "project_name": "Your new project name - Enw eich prosiect newydd",
     "project_repo_name": "{{ cookiecutter.project_name.lower().replace(' ', '_') }}",
-    "project_description": "Brief project description/Disgrifiad byr o'r prosiect",
-    "project_team": "Project team name e.g. DKR Data Science/Enw tîm y prosiect e.e. Gwyddor Data",
-    "contact_email": "Project contact email address/Cyfeiriad e-bost cyswllt y prosiect",
+    "project_description": "Brief project description - Disgrifiad byr o'r prosiect",
+    "project_team": "Project team name e.g. DKR Data Science - Enw tîm y prosiect e.e. Gwyddor Data",
+    "contact_email": "Project contact email address - Cyfeiriad e-bost cyswllt y prosiect",
+    "include_weekly_data_exp_log_template": "1 = Yes-Ie, 0 = No-Na"
```
Comment (phwrwalms) [inline cookiecutter.json:7]: include_user_story_templates has been deleted here, but is still used by the code?

File: {{cookiecutter.project_repo_name}}/src/{{cookiecutter.project_repo_name}}/example_module.py
Comment (phwrwalms) [inline {{cookiecutter.project_repo_name}}/src/{{cookiecutter.project_repo_name}}/example_module.py:1]: Why is this file deleted?

File: {{cookiecutter.project_repo_name}}/.github/ISSUE_TEMPLATES/mater-stori-defnyddiwr-safonol.md
Comment (phwrwalms) [inline {{cookiecutter.project_repo_name}}/.github/ISSUE_TEMPLATES/mater-stori-defnyddiwr-safonol.md:1]: I think the folder name should be ISSUE_TEMPLATE (https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-templates) to be auto-recognised by GitHub? 

ISSUE_TEMPLATES might work but not sure. I've looked at a couple of big projects and they seem to have .github/ISSUE_TEMPLATE

File: {{cookiecutter.project_repo_name}}/.github/templates/{% if cookiecutter.include_user_story_templates == 'yes' %}mater-stori-defnyddiwr-safonol.md{% endif %}
Comment (phwrwalms) [reply {{cookiecutter.project_repo_name}}/.github/templates/{% if cookiecutter.include_user_story_templates == 'yes' %}mater-stori-defnyddiwr-safonol.md{% endif %}:1]: Thanks for the fix :)

File: {{cookiecutter.project_repo_name}}/.github/ISSUE_TEMPLATES/mater-stori-defnyddiwr-safonol.md
Comment (phwrwalms) [reply {{cookiecutter.project_repo_name}}/.github/ISSUE_TEMPLATES/mater-stori-defnyddiwr-safonol.md:1]: Thanks :)

**Review by phwrwalms — APPROVED** [review]
